template/funcs.go

```diff
@@ -47,11 +47,22 @@ func datacentersFunc(b *Brain, used, missing *dep.Set) func() ([]string, error)
 }
 
 // executeTemplateFunc executes the given template in the context of the
-// parent. This can be used for nested template definitions.
-func executeTemplateFunc(t *template.Template) func(string) (string, error) {
-	return func(s string) (string, error) {
+// parent. If an argument is specified, it will be used as the context instead.
+// This can be used for nested template definitions.
+func executeTemplateFunc(t *template.Template) func(string, ...interface{}) (string, error) {
+	return func(s string, data ...interface{}) (string, error) {
+		var dot interface{}
+		switch len(data) {
+		case 0:
+			dot = nil
+		case 1:
+			dot = data[0]
+		default:
+			return "", fmt.Errorf("executeTemplate: wrong number of arguments, expected 1 or 2"+
+				", but got %d", len(data)+1)
```

Comment:
I think the +1 is correct because `len(data)` does not account for the previous (required) argument for the template name.

    {{ executeTemplate "t" }}         => the original usage
    {{ executeTemplate "t" .ctx }}  => the augmented case
    {{ executeTemplate "t" .ctx .something_else }}  
            => error "wrong number of arguments, expected 1 or 2, but got 3"